setup.py

```diff
@@ -116,17 +107,17 @@
         "Topic :: Scientific/Engineering :: Artificial Intelligence",
         "Topic :: Software Development :: Libraries :: Python Modules",
         "License :: OSI Approved :: MIT License",
-        "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
+        # "Programming Language :: Python :: 3.10",
         "Operating System :: POSIX :: Linux",
     ],
     extras_require=extras_require,
     keywords="recommendations recommendation recommenders recommender system engine "
     "machine learning python spark gpu",
     install_requires=install_requires,
     package_dir={"recommenders": "recommenders"},
-    python_requires=">=3.6, <3.10",
+    python_requires=">=3.8, <=3.9",
```

Comment:
```suggestion
    python_requires=">=3.6",
```
@SimonYansenZhao we had a discussion in the reco meeting. There are a couple of things we are proposing:
1. leave python requires with >= 3.6 so people can install recommenders with any version of python. Right now, for example, people can't install the pypi package on 3.10  Then we say that we officially only support 3.8-3.10. With that we avoid issues like https://github.com/recommenders-team/recommenders/issues/1928 where people can't install recommenders on versions that are not in the requires, and at the same time, we officially only tests the versions we support. 
2. Maybe it would be easier to split this PR into 2, first we remove the support of 3.7, and then in another PR we add 3.10.

What do you think?

Reply:
OK.